controllers/remove_process_groups.go

```diff
@@ -138,108 +141,112 @@ func (u removeProcessGroups) reconcile(ctx context.Context, r *FoundationDBClust
 	return nil
 }
 
-func removeProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster, processGroupID fdbv1beta2.ProcessGroupID) error {
-	listOptions := internal.GetSinglePodListOptions(cluster, processGroupID)
-	pods, err := r.PodLifecycleManager.GetPods(ctx, r, cluster, listOptions...)
-	if err != nil {
-		return err
-	}
+func removeProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster, processGroup *fdbv1beta2.ProcessGroupStatus) error {
+	podName := processGroup.GetPodName(cluster)
+	var deletionError error
 
-	if len(pods) == 1 && pods[0].DeletionTimestamp.IsZero() {
-		err = r.PodLifecycleManager.DeletePod(ctx, r, pods[0])
+	pod, err := r.PodLifecycleManager.GetPod(ctx, r, cluster, podName)
+	if err == nil && pod.DeletionTimestamp.IsZero() {
+		deletionError = r.PodLifecycleManager.DeletePod(ctx, r, pod)
 		if err != nil {
-			return err
+			deletionError = fmt.Errorf("could not delete Pod: %w", err)
 		}
-	} else if len(pods) > 1 {
-		return fmt.Errorf("multiple pods found for cluster %s, processGroupID %s", cluster.Name, processGroupID)
 	}
 
+	// TODO(johscheuer): https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1638
 	pvcs := &corev1.PersistentVolumeClaimList{}
-	err = r.List(ctx, pvcs, listOptions...)
+	err = r.List(ctx, pvcs, internal.GetSinglePodListOptions(cluster, processGroup.ProcessGroupID)...)
 	if err != nil {
 		return err
 	}
 	if len(pvcs.Items) == 1 && pvcs.Items[0].DeletionTimestamp.IsZero() {
 		logr.FromContextOrDiscard(ctx).V(1).Info("Deleting pvc", "name", pvcs.Items[0].Name)
 		err = r.Delete(ctx, &pvcs.Items[0])
 		if err != nil {
-			return err
+			pvcError := fmt.Errorf("could not delete PVC: %w", err)
+			if deletionError == nil {
+				deletionError = pvcError
+			} else {
+				deletionError = fmt.Errorf("previous error: %w, %w", deletionError, pvcError)
+			}
 		}
 	} else if len(pvcs.Items) > 1 {
-		return fmt.Errorf("multiple PVCs found for cluster %s, processGroupID %s", cluster.Name, processGroupID)
+		return fmt.Errorf("multiple PVCs found for cluster %s, processGroupID %s", cluster.Name, processGroup.ProcessGroupID)
```

Comment:
do you think we should also include `deletionError` here ? Otherwise if all the previous error conditions are true those errors will be ignored completely.

Reply:
We probably can ignore that here, this code will be changed in a few weeks anyway to make use of the same get mechanism as for service and Pod.